test/finch_test.exs

```diff
@@ -379,14 +379,15 @@ defmodule FinchTest do
 
       # 2KB of data
       req_body = List.duplicate(125, 2_000)
```

Comment:
@the-mikedavis I wonder if it needs to be increased to 2MB as in the binary version of this test?

Reply:
so I think this bug is actually more about mis-handling iodata than the iodata exceeding the window size (hence why you hit this with only 2KB when the default window is ~64KB if I remember correctly)

it probably makes sense to increase it to 2MB as you say though because that should effectively test both that iodata is handled correctly, and that window sizes are respected when using iodata